sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -139,6 +139,12 @@ def done(self):
       self._state = self.DONE_STATE
 
 
+class _ListBuffer(list):
+  """Used to support parititioning of a list."""
+  def partition(self, n):
+    return [self[k::n] for k in range(n)] if self else [[] for _ in range(n)]
```

Comment:
You can drop the `if self else [[] for _ in range(n)]`, the slicing will work fine on an empty list as well.